crates/nu-data/src/base.rs

```diff
@@ -64,20 +64,34 @@ pub fn select_fields(obj: &Value, fields: &[String], tag: impl Into<Tag>) -> Val
     out.into_value()
 }
 
-pub fn reject_fields(obj: &Value, fields: &[String], tag: impl Into<Tag>) -> Value {
+pub fn reject_fields(
+    obj: &Value,
+    fields: &[Tagged<String>],
+    tag: impl Into<Tag>,
+) -> Result<Value, ShellError> {
     let mut out = TaggedDictBuilder::new(tag);
 
     let descs = obj.data_descriptors();
 
+    for field in fields {
+        if descs.iter().all(|desc| desc != &field.item) {
+            return Err(ShellError::labeled_error(
+                "Invalid column",
```

Comment:
The original for loop below this does not give access to each field, which we need to grab the tag for the error, so I had to basically those original nested loops and flip them inside out in order to get the check to work.